salt/cni/update-pre-shutdown.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/sh
+
+NODE_ID="$1"
+EXTERNAL_IP="$2"
+
+FLANNEL_STATE_FILE="/run/flannel/subnet.env"
+
+##########################################################
+
+log() { echo "[CNI migration]: $1 " ; logger -t "cni-migration" "$1" ; }
+
+exit_changes() {
+	log "$2"
+	echo  # an empty line here so the next line will be the last.
+	echo "changed=$1 comment='"$2"'"
+	exit 0
+}
+
+get_node_cidr() {
+	kubectl get no "$NODE_ID" --template="{{.spec.podCIDR}}"
+}
+
+patch_node() {
+	kubectl patch node $NODE_ID -p $@ 2>/dev/null
+}
+
+##########################################################
+
+log "migrating $NODE_ID CIDR"
+
+[ -e "$FLANNEL_STATE_FILE" ] || exit_changes "no" "no flannel state file found"
+source $FLANNEL_STATE_FILE
+old_node_cidr=$(echo "$FLANNEL_SUBNET" | sed -e "s/\.1\//\.0\//g")
```

Comment:
`sed -e "s#\.1/#\.0/#g"` is kind of more readable, but it's fine, took me a while to find the trailing `/` to make sure that we were only replacing the last octet.